### ISSUE TITLE
False positive error using embedFile with relative path

### ISSUE THREAD

**Issue by chris-martin** [flycheck/flycheck-haskell#53]
I'm [`Data.FileEmbed`](http://haddock.stackage.org/lts-3.22/file-embed-0.0.9/Data-FileEmbed.html) like this:

    $(embedFile "foo")

This compiles fine, but flycheck reports the file `foo` doesn't exist. If I specify an absolute path, or if I change `foo` to `../../../foo`, then it doesn't report an errror.

Is there maybe some command that flycheck is running from the directory in which the checked `.hs` file resides, that should be running instead at the top of the project where the stack/cabal files reside?

**Comment by lunaryorn** [comment]
No, there isn't yet.  This requires deeper changes to Flycheck, see https://github.com/flycheck/flycheck/pull/813.  It'll come, some day, but not any time soon.  It's a larger and deeper change, it requires careful testing, and that takes time that I don't have currently.

**Comment by lunaryorn** [comment]
I'm closing this issue, please subscribe to flycheck/flycheck#813 to keep track of this.  Thank you for your feedback.

**Comment by xenog** [comment]
The Flycheck change seems to have happened, but this still doesn't work for me.

**Comment by sergv** [comment]
@xenog Can you provide a reproducible test case?

**Comment by xenog** [comment]
@sergv, clone [this repository](https://github.com/haskoin/haskoin) in Spacemacs with the `haskell` and `syntax-checking` layers and then open [this file](https://github.com/haskoin/haskoin/blob/master/haskoin-wallet/Network/Haskoin/Wallet/Settings.hs). You should see this error near `embedFile`:

```
   109  12 error           • Exception when trying to run compile-time code:
     config/config.yml: openBinaryFile: does not exist (No such file or directory)
   Code: embedFile "config/config.yml"
 • In the untyped splice: $(embedFile "config/config.yml") (haskell-stack-ghc)
```

**Comment by sergv** [comment]
@xenog I have reproduced the problem and I think I know what's causing the issue. The issue with current directory was resolved and now ghc is being executed from the directory that contains `stack.yaml` file, which in this case in the root of the project. However, file path is specified relative to root of *subproject*, thus the `does not exist` error. However, I cannot think from the top of my head of any reasonable strategy to deal with this issue...
